src/stratigraphy/benchmark/metrics.py

```diff
@@ -53,23 +53,62 @@ def to_dataframe(self, name: str, fn: Callable[[Metrics], float]) -> pd.DataFram
         return series.to_frame(name=name)
 
 
-class DatasetMetricsCatalog:
+class OverallMetricsCatalog:
     """Keeps track of all different relevant metrics that are computed for a dataset."""
 
-    def __init__(self):
-        self.metrics: dict[str, DatasetMetrics] = {}
+    layer_metrics: OverallMetrics = None
+    depth_interval_metrics: OverallMetrics = None
+    de_layer_metrics: OverallMetrics = None
+    de_depth_interval_metrics: OverallMetrics = None
+    fr_layer_metrics: OverallMetrics = None
+    fr_depth_interval_metrics: OverallMetrics = None
+    groundwater_metrics: OverallMetrics = None
+    groundwater_depth_metrics: OverallMetrics = None
+
+    def set_layer_metrics(self, metrics: OverallMetrics):
+        """Set the layer metrics."""
+        self.layer_metrics = metrics
+
+    def set_depth_interval_metrics(self, metrics: OverallMetrics):
+        """Set the depth interval metrics."""
+        self.depth_interval_metrics = metrics
+
+    def set_de_layer_metrics(self, metrics: OverallMetrics):
```

Comment:
Having one setter per language might be a bit too much. We could think of passing the language as a parameter. This would save us the declaration of four functions.

Reply:
Is is even necessary to define these setter methods at all? It's not like the class' instance variables are private.

Reply:
Yes, that is actually true. I will remove and replace this.